components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -327,27 +337,58 @@ export default function RepositoryFinder({
-                // ENT-780
+            if (
+                searchString.length > 0 &&
+                searchString.length < 3 &&
+                authProviders.data?.some((p) => p.type === AuthProviderType.GITLAB) &&
+                !onlyConfigurations
+            ) {
+                // add an element that tells the user that GitLab only does exact searches for short queries
                 result.push({
-                    id: "azure-devops",
+                    id: "gitlab",
                     element: (
                         <div className="text-sm text-pk-content-tertiary flex items-center">
                             <Exclamation2 className="w-4 h-4 mr-2" />
-                            <span>Azure DevOps doesn't support repository searching.</span>
+                            <span>Search is too short for searching on GitLab.</span>
+                        </div>
+                    ),
+                    isSelectable: false,
+                });
+            }
+
+            const setupProvidersWithoutPathSearchSupport = usedProviders.filter((p) =>
+                ["Bitbucket", "Bitbucket Server", "GitLab"].includes(p),
+            );
+            if (
+                searchString.length > 1 &&
+                setupProvidersWithoutPathSearchSupport.length > 0 &&
+                searchString.includes("/")
+            ) {
+                result.push({
+                    id: "whole-path-matching-unsupported",
+                    element: (
+                        <div className="text-sm text-pk-content-tertiary flex items-center">
+                            <Exclamation2 className="w-4 h-4 mr-2" />
+                            <span>
+                                {usedProviders
+                                    ? conjunctScmProviders(setupProvidersWithoutPathSearchSupport)
+                                    : "Some providers"}{" "}
+                                only support searching by repository name, not full paths.
+                            </span>
                         </div>
                     ),
                     isSelectable: false,
                 });
             }
 
-            if (searchString.length < 3) {
-                // add an element that tells the user to type more
+            if (searchString.length > 0 && authProviders.data?.some((p) => p.type === AuthProviderType.AZURE_DEVOPS)) {
+                // CLC-780
                 result.push({
-                    id: "not-searched",
+                    id: "azure-devops",
                     element: (
-                        <div className="text-sm text-pk-content-tertiary">
-                            Please type at least 3 characters to search.
+                        <div className="text-sm text-pk-content-tertiary flex items-center">
+                            <Exclamation2 className="w-4 h-4 mr-2" />
+                            <span>Azure DevOps doesn't support repository searching.</span>
```

Comment:
Should we show this message if `onlyConfigurations === true`? (lagging at bit of context, just trying to understand from the code in this PR).

Reply:
Totally missed this, thanks for pointing it out. We should only show these messages when `onlyConfigurations === false`, since only then do SCM-provided repos come into play.

Reply:
Added in [`c50559b` (#20415)](https://github.com/gitpod-io/gitpod/pull/20415/commits/c50559bf614ff590c221e3f986fe75efbef897ae).